doc/manual_experimental.rst

```diff
@@ -230,65 +274,32 @@ scope. Therefore, the following will *fail to compile:*
 
   a()
 
+This feature will likely be replaced with a better solution to remove
+the need for forward declarations.
 
-Named argument overloading
-==========================
 
-Routines with the same type signature can be called differently if a parameter
-has different names. This does not need an `experimental` switch, but is an
-unstable feature.
-
-.. code-block:: Nim
-
-  proc foo(x: int) =
-    echo "Using x: ", x
-  proc foo(y: int) =
-    echo "Using y: ", y
-
-  foo(x = 2)
-  # Using x: 2
-  foo(y = 2)
-  # Using y: 2
-
-
-Do notation
-===========
+Automatic dereferencing
+=======================
 
-As a special more convenient notation, proc expressions involved in procedure
-calls can use the `do` keyword:
+Automatic dereferencing is performed for the first argument of a routine call.
+This feature has to be enabled via `{.experimental: "implicitDeref".}`:
 
 .. code-block:: nim
 
-  sort(cities) do (x,y: string) -> int:
-    cmp(x.len, y.len)
-
-  # Less parenthesis using the method plus command syntax:
-  cities = cities.map do (x:string) -> string:
-    "City of " & x
-
-  # In macros, the do notation is often used for quasi-quoting
-  macroResults.add quote do:
-    if not `ex`:
-      echo `info`, ": Check failed: ", `expString`
-
-`do` is written after the parentheses enclosing the regular proc params.
-The proc expression represented by the do block is appended to them.
-In calls using the command syntax, the do block will bind to the immediately
-preceding expression, transforming it in a call.
-
-`do` with parentheses is an anonymous `proc`; however a `do` without
-parentheses is just a block of code. The `do` notation can be used to
-pass multiple blocks to a macro:
+  {.experimental: "implicitDeref".}
 
-.. code-block:: nim
+  type
+    NodeObj = object
+      # ...
+    Node = ref NodeObj
 
-  macro performWithUndo(task, undo: untyped) = ...
+  proc depth(x: NodeObj): int = ...
 
-  performWithUndo do:
-    # multiple-line block of code
-    # to perform the task
-  do:
-    # code to undo it
+  var
```

Comment:
I'd just write `let n = new(Node)` instead.

Reply:
`new(Node)` just expands to `(var n: Node; new(n); n)`. `let n = Node()` is the preferred way I believe?

Reply:
Oh yeah, that's even better.